src/lib/integration_test_cloud_engine/coverage_manager.ml

```diff
@@ -0,0 +1,55 @@
+(* Submodule responsible for managing test coverage files. Responsible for code coverage raw files generatation from pods.
+   Requires tailored mina images with instrumentation and BISECT_SIGTEM env variable set.
+   Function traverse all applicable pods (those which has mina process) and kills them.
+   This operation dumps raw bissect files (.coverage) and downloads it to gcloud bucket.
+   WARNING: operations of test coverage retrieval is destructive (as it kills mina processes).
+*)
+open Integration_test_lib
+open Core
+
+module Coverage_manager = struct
+  type t = { logger : Logger.t }
+
+  let create ~logger = { logger }
+
+  let download_coverage_data_from_nodes t ~nodes =
+    let open Kubernetes_network in
+    let root = "/root/.mina-config" in
+    let open Malleable_error.Let_syntax in
+    [%log' info t.logger] "Generating test coverage data..." ;
+    let coverage_files =
+      nodes |> Map.to_alist
+      |> Malleable_error.List.fold ~init:[] ~f:(fun acc (_id, pod) ->
```

Comment:
You probably want to actually do this in parallel, as this is all IO that you don't want to block on, and you will see a significant speedup of this step.

You can do that by building a list of malleable error monads first, and then folding them together at the end. Each malleable error will run in parallel in the background and the final fold will wait for each value to be ready.